src/Symfony/Component/Console/Descriptor/MarkdownDescriptor.php

```diff
@@ -75,15 +87,16 @@ protected function describeInputOption(InputOption $option, array $options = [])
         );
     }
 
+    /**
+     * @return void
+     */
     protected function describeInputDefinition(InputDefinition $definition, array $options = [])
     {
         if ($showArguments = \count($definition->getArguments()) > 0) {
             $this->write('### Arguments');
             foreach ($definition->getArguments() as $argument) {
                 $this->write("\n\n");
-                if (null !== $describeInputArgument = $this->describeInputArgument($argument)) {
-                    $this->write($describeInputArgument);
-                }
+                $this->describeInputArgument($argument);
```

Comment:
Yes, before 5.4, this even was `$this->write($this->describeInputArgument($argument))`. The 5.4 type clean-up already resulted in adding an if statement and this cleanup completely rules out any return value.

In very old Symfony versions, the describe methods returned an array of strings instead of using `OutputInterface` directly. Seems like we didn't correctly clean-up during that refactor.